docs/architecture/README.md

```diff
@@ -3,111 +3,184 @@ order: 3
+The **Worker** is where all tasks are run. A Worker will receive a task from the Executor and will process it. As tasks can be nearly anything (heavy computations, simple api calls, etc...), the Worker is essentially a Thread Pool where you can configure the amount of threads you need.
 
-You can also scale Workers as necessary, and have many instances on multiple servers (within the limits of the partitions count on Kafka), each with its own Thread Pool. If a Worker is dead, the Executor will detect it and resubmit their current task run to another Worker.
+You can scale Workers as necessary, and have many instances on multiple servers, each with its own Thread Pool.
+
+As the worker will execute the tasks, it needs to have access to any external services used by your tasks (database, REST services, message broker, ...).
 
 ### Scheduler
-The Scheduler will handle most of the [Triggers](../developer-guide/triggers) (except the [Flow Trigger handle by the Executor](../developer-guide/triggers/flow.md)). It will continuously watch all triggers and, if all the conditions are met, will trigger an Execution (submitted to the Executor).
 
-Internally, a Scheduler has two Thread Pools:
+The **Scheduler** will handle most of the [Triggers](../developer-guide/triggers) (except the [Flow Triggers handled by the Executor](../developer-guide/triggers/flow.md)). It will continuously watch all triggers and, if all the conditions are met, will trigger an execution (submit it to the Executor).
 
+Internally, a Scheduler has two Thread Pools:
 - **ScheduledExecutorService**: This Thread Pool will tick every second and try to find an available job.
 - **CachedThreadPool**: This Thread Pool will deal with all other scheduler computational needs: detecting files, executing queries, fetching queues, ...
 
 ::: warning
 By default, Kestra will handle all dates based on your system timezone. You can change the timezone with [JVM options](../administrator-guide/configuration/others#jvm-configuration).
 :::
 
-::: warning
-With MySQL or PostgreSQL as database, you can't scale the replicas count for the scheduler, and you must have only **one** instance for the whole cluster.
-:::
+Some triggers may require to have access to external services (triggers based on database query or message subscription for example). If you use such triggers the Scheduler will need to have access to the required external services as it executes the trigger.
 
 ### Indexer
-The Indexer will watch for many Kafka topics and will simply copy the content to [Repositories](#repositories). This will save all the executions & flows to ElasticSearch providing Kestra with a rich UI.
+
+The **Indexer** is an optional component only needed when using Kafka and Elasticsearch. It will watch for many Kafka topics (like flows and executions) and will index there content to Elasticsearch [Repositories](#repositories).
```

Comment:
```suggestion
The **Indexer** is an optional component only needed when using Kafka and Elasticsearch. It will watch for many Kafka topics (like flows and executions) and will index their content to Elasticsearch [Repositories](#repositories).
```